### DISCUSSION TITLE
Recommendation Engine Ideas

### DISCUSSION THREAD

**Discussion by dbaslan** [SWE574-Fall2023-Group1/SWE574-Fall2023-G1#149] [Ideas]
@SWE574-Fall2023-Group1/all  Alright folks, we need to design the backend for the recommendation system. What should we pay attention to? Let's share our ideas. Title, text, memory date, location, tags and so forth...

**Comment by ckertam** [comment]
Firstly there will be 3 key elements
- Flags (Those will be extracted from title,content and tag)
- Time info
- Location info

Flag Logic will work as below
- While creating a story there will be a running nlp model so it will extract 5 words for each story. 
- Title, content and tag will be searched for extracting flags. 

To recommend stories the system will work as below
- There will be a recommendation table.
- It will keep stories and their compatibility with user's interest. 
- Such as a user likes a story. After that the system will work and add stories to recommendation database table. While adding stories' there will be relation order. As I mentioned there will be 5 flags in each story, according to matching flag number the stories will be sorted. 
- Also, the location and time info will be used as normal search logic. 
- So there will be 3 stories to recommend to user. 
- One will recommended by flags, the other 2 will be recommended by location and time info. 
- Besides there will be is_shown and is_clicked fields.
- If a user once see a recommended story it's is_shown field will turn to true. If a story is clicked is_clicked field will turn to true.
- According to those fields  the search recommendation show logic will be ordered so that the system will decide to show which story first. Such as a story with is_shown field with a true value will not shown if there is a false value story. 


The system is open to improve. Let's discuss it further!

**Reply by ckertam** [reply]
After some chat with professor we decided to make little changes. 

For story flag part it will be only created via content and will use keyword related nlp models. 

For tags, we will improve our tags so that they will consist of a semantic layer too.
